### ISSUE TITLE
Line number reported for invalid argument type can be confusing

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#4231]
Here it looks like mypy is complaining about the first `f` call, but it's actually complaining about the second one:

```py
class A:
    def f(self, x: int) -> 'A':
        return self

a: int
b: str

A().f(  # Argument 1 to "f" of "A" has incompatible type "str"; expected "int"
    a
).f(  # Why no error here?
    b
)
```

I actually lost a fair amount of time because of this issue. Maybe we should give the line number of the argument expression with an incompatible type.

**Comment by hauntsaninja** [comment]
Can't repro on master (error is on line 11)
